lib/spack/spack/database.py

```diff
@@ -393,7 +401,7 @@ def _add(self, spec, path, directory_layout=None):
 
         else:
             self._data[key] = InstallRecord(spec, path, True)
-            for dep in spec.dependencies.values():
+            for dep in spec.dependencies('link'):
```

Comment:
Should this be *only* link?  I can really only see pure build deps getting dropped here, as they are actually no longer needed by dependents after the build step.  `link` *and* `run`, however, will still be needed by installed dependents, so shouldn't this increment for both `link` and `run` deptypes? If you take the run dep away, Spack can't guarantee that the thing will actually run.